Mathlib/Topology/MetricSpace/Closeds.lean

```diff
@@ -235,6 +238,16 @@ theorem NonemptyCompacts.ToCloseds.isUniformEmbedding :
     IsUniformEmbedding (@NonemptyCompacts.toCloseds α _ _) :=
   Isometry.isUniformEmbedding fun _ _ => rfl
 
+lemma NonemptyCompacts.isClosed_subsets_of_isClosed (hs : IsClosed s) :
+    IsClosed {A : NonemptyCompacts α | (A : Set α) ⊆ s} := by
+  refine isClosed_of_closure_subset fun E hE x hx ↦ ?_
+  suffices x ∈ closure s by rwa [hs.closure_eq] at this
+  rw [mem_closure_iff] at hE ⊢
+  intro ε hε
+  obtain ⟨C, hC, hEC⟩ := hE ε hε
+  obtain ⟨y, hy, hxy⟩ := exists_edist_lt_of_hausdorffEdist_lt hx hEC
+  exact ⟨y, hC hy, hxy⟩
+
```

Comment:
The statement right above tells you (in particular) that the forgetful map from nonempty compacts to closeds is continuous (this should be stated explicitly), so I think you can directly reuse the closed version.